nipype/interfaces/fsl/utils.py

```diff
@@ -262,6 +262,70 @@ def _format_arg(self, name, trait_spec, value):
         return super(Smooth, self)._format_arg(name, trait_spec, value)
 
 
+class SliceInputSpec(FSLCommandInputSpec):
+    in_file = File(exists=True, argstr="%s", position=0, mandatory=True,
+                   desc="input filename")
+    out_base_name = traits.Str(argstr="%s", position=1, desc="outputs prefix")
+
+
+class SliceOutputSpec(TraitedSpec):
+    out_files = OutputMultiPath(File(exists=True))
+
+
+class Slice(FSLCommand):
+    """Use fslslice to split a 3D file into lots of 2D files (along z-axis).
+
+
+    Examples
+    --------
+
+    >>> from nipype.interfaces.fsl import Slice
+    >>> slice = Slice()
+    >>> slice.inputs.in_file = 'functional.nii'
+    >>> slice.inputs.out_base_name = 'sl'
+    >>> slice.cmdline
+    'fslslice functional.nii sl'
+
+
+    """
+
+    _cmd = 'fslslice'
+    input_spec = SliceInputSpec
+    output_spec = SliceOutputSpec
+
+    def _list_outputs(self):
+        """Create a Bunch which contains all possible files generated
+        by running the interface.  Some files are always generated, others
+        depending on which ``inputs`` options are set.
+
+        Returns
+        -------
+
+        outputs : Bunch object
+            Bunch object containing all possible files generated by
+            interface object.
+
+            If None, file was not generated
+            Else, contains path, filename of generated outputfile
+
+        """
+        outputs = self._outputs().get()
+        ext = Info.output_type_to_ext(self.inputs.output_type)
+        if isdefined(self.inputs.out_base_name):
+            outbase = self.inputs.out_base_name + '_slice_*'
+            out_dir = os.getcwd()
+        else:
+            out_dir = os.path.dirname(self.inputs.in_file)
+            in_base, _ = os.path.splitext(self.inputs.in_file)
+            if '.gz' in self.inputs.in_file:
+                in_base, _ = os.path.splitext(in_base)
+
+            outbase = in_base + '_slice*'
+        outputs['out_files'] = sorted(glob(os.path.join(out_dir,
+                                                        outbase + ext)))
```

Comment:
This doesn't account for if `out_base_name` is an absolute path. You can simplify this a bit, too, using `fname_presuffix`:

```Python
suffix = '_slice_*' + ext
if isdefined(self.inputs.out_base_name):
    fname_template = os.abspath(self.inputs.out_base_name + suffix)
else:
    fname_template = fname_presuffix(self.inputs.in_file, suffix=suffix, use_ext=False)

outputs['out_file'] = sorted(glob(fname_template))
```